src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs

```diff
@@ -336,12 +339,11 @@ private static string GetValueOrThrowMissingEnvVar(Func<string> getValue, string
 
         public static string GetDotNetLocation()
         {
-            var path = Environment.GetEnvironmentVariable("PATH");
+            var path = EnvironmentVariableProvider.GetEnvironmentVariable("PATH");
             var isWindows = RuntimeEnvironmentHelper.IsWindows;
-            var splitChar = isWindows ? ';' : ':';
             var executable = isWindows ? DotNetExe : DotNet;
 
-            foreach (var dir in path.Split(splitChar))
+            foreach (var dir in path.Split(Path.PathSeparator))
```

Comment:
This is not the same, and I'm a bit concerned why no test is failing because of it

Reply:
This should be the same, `Path.PathSeparator` is `;` on Windows and `:` on non-Windows.  Are you thinking I used `Path.DirectorySeparator`?  That's different.

https://learn.microsoft.com/en-us/dotnet/api/system.io.path.pathseparator?view=net-8.0#remarks

Reply:
🤦

How do I make the 🤦 bigger, because it's a big 🤦 moment.

Yes, I was thinking DirectorySeparator 🤦